### PR TITLE
Fix PRB

### PR DESCRIPTION
**Pull request by xrigau**
Attempt to make all the tests pass by using the up-to-date version of some libs and update the usage of old methods.

I also extracted the server URL mapping to its own class and added a test for it

### CODE REVIEW

**Comment by xrigau** [comment]
Retest this please

**Comment by xrigau** [comment]
Retest this

File: angular2-client/src/app/server-url.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  mapUrl(): (config: Config) => string {
+    return (config: Config) => {
+      let host: string = this.isLocal() ? 'http://localhost' : config.serverUrl;
+      return host + ':' + config.port;
+    };
+  }
+
+  private isLocal() {
+    return window.location.hostname === 'localhost' || window.location.hostname === '127.0.0.1';
```
Comment (frapontillo) [inline angular2-client/src/app/server-url.service.ts]: HAX! Why don't you use `src/environments/environment.(dev|prod|mock).ts`?

**Comment by xrigau** [comment]
Retest this

File: angular2-client/src/app/server-url.service.ts (hunk 1 above)
Comment (xrigau) [inline angular2-client/src/app/server-url.service.ts]: because I'm a noob 🌮

File: angular2-client/src/app/server-url.service.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { Config } from './Config';
+import { environment } from './environments/environment';
+
+@Injectable()
+export class ServerUrlService {
+
+  mapUrl(): (config: Config) => string {
+    return (config: Config) => {
+      let host: string = environment.production ? config.serverUrl : 'http://localhost';
```
Comment (frapontillo) [inline angular2-client/src/app/server-url.service.ts]: 💯

Comment (xrigau) [inline angular2-client/src/app/server-url.service.ts]: @frapontillo 👀

File: angular2-client/src/app/config.service.spec.ts [hunk 3]
```diff
@@ -1,40 +1,60 @@
...
+    });
   });
 
-  beforeEach(inject([MockBackend], (_mockBackend_) => {
+  beforeEach(inject([MockBackend, ServerUrlService], (_mockBackend_, _serverUrlService_) => {
     mockBackend = _mockBackend_;
+    serverUrlService = _serverUrlService_;
   }));
 
+  const givenLocalServer = () => {
```
Comment (xrigau) [inline angular2-client/src/app/config.service.spec.ts]: @frapontillo now that we're using `environment`, do you know if there's a better way of doing this? (i.e. forcing the environment somehow or something like that)

Comment (frapontillo) [inline angular2-client/src/app/config.service.spec.ts]: Nah. This is good and shiny.

Comment (xrigau) [inline angular2-client/src/app/config.service.spec.ts]: 👍  great!

**Pull request merged**